### PR TITLE
Enhancement/make queue parameter required for connector

### PR DESCRIPTION
**Pull request by evelyn9191**

### CODE REVIEW

File: elisctl/connector.py [hunk 1]
```diff
@@ -34,6 +38,16 @@ def create_command(
...
     with ELISClient(context=ctx.obj) as elis:
+        if _is_empty(queue_ids):
+            queues = elis.get_queues()
+            if len(queues) == 1:
+                try:
+                    queue_ids = tuple(int(queue["id"]) for queue in queues)
+                except KeyError:
+                    raise click.ClickException("Could not get queue ID.")
+            else:
+                raise click.ClickException("Queue ID(s) must be specified.")
```
Comment (KPilnacek) [inline elisctl/connector.py]: consider rather using something like:
```python
        if queue_ids:
            queue_urls = [q['url'] for q in elis.get_queues() if int(q['id']) in queue_ids]
        else:
            queue_urls = [elis.get_queue()['url']]
```
in order to decrease the number of requests and simplify the code. The for loop which follows this part would be also replaced by this code.

File: elisctl/connector.py [hunk 2]
```diff
@@ -140,3 +156,10 @@ def delete_command(ctx: click.Context, id_: str) -> None:
 def _generate_token():
     alphabet = string.ascii_letters + string.digits
     return "".join(secrets.choice(alphabet) for _ in range(36))
+
+
+def _is_empty(attr: Any):
```
Comment (KPilnacek) [inline elisctl/connector.py]: Please explain `_is_empty` function. It returns `False` when `queue_ids == (1,2)`. That doesn't sound to me as correct result of function that should check if sth is empty.

File: elisctl/connector.py [hunk 3]
```diff
@@ -48,7 +62,9 @@ def create_command(
             params=params,
             asynchronous=asynchronous,
         )
-        click.echo(f"{response['id']}, {response['name']}")
+        click.echo(
+            f"{response['id']}, {response['name']}, attached to the queue(s) {response['queues']}"
```
Comment (KPilnacek) [inline elisctl/connector.py]: I'd rather keep the outputs of elisctl programmatically parsable (i.e. leave out "attached to the queue(s)").

File: elisctl/connector.py [hunk 4]
```diff
@@ -16,7 +16,11 @@ def cli() -> None:
 
 @cli.command(name="create", help="Create a connector object.")
 @argument.name
-@option.queue
+@option.queue(
+    element="connector",
```
Comment (KPilnacek) [inline elisctl/connector.py]: element="connector", not necessary as it is overwritten in help anyway

File: elisctl/option.py [hunk 5]
```diff
@@ -44,14 +44,18 @@ def workspace_id(command: Optional[Callable] = None, **kwargs):
...
 
-queue = click.option(
-    "-q",
-    "--queue-id",
-    "queue_ids",
-    type=int,
-    multiple=True,
-    help="Queue IDs, which the user will be associated with.",
-)
+def queue(command: Optional[Callable] = None, element: Optional[str] = "object", **kwargs):
```
Comment (KPilnacek) [inline elisctl/option.py]: consider renaming `element` to sth like `related_object`

**Review by KPilnacek — CHANGES_REQUESTED** [review]

File: elisctl/connector.py (hunk 2 above)
Comment (evelyn9191) [reply elisctl/connector.py]: That sounds correct to me. If <code>queue_ids == (1,2)</code>, then the tuple is not empty and <code>_is_empty</code> should return <code>False</code>.<code>_is_empty</code> checks if the value equals <code>None</code> or (particularly in this case) <code>()</code> - in such case it returns <code>True</code>. It is a variation on <code>if len(x) == 0</code>.

Comment (KPilnacek) [reply elisctl/connector.py]: Oh, sorry, my bad. It's correct.

However, I'd rather do something like `return not attr`. Actually, I'd remove the function completely and only use the `not` wherever necessary. To me, the current solutions is confusing.

File: elisctl/connector.py (hunk 1 above)
Comment (KPilnacek) [reply elisctl/connector.py]:
```python
        if not queue_ids:
            queue_urls = [elis.get_queue()['url']]
        else:
            queue_urls = []
            for id_ in queue_ids:
                queue_dict = elis.get_queue(id_)
                if queue_dict:
                    queue_urls.append(queue_dict["url"])
```

File: elisctl/lib/api_client.py [hunk 6]
```diff
@@ -256,6 +256,8 @@ def get_queue(
                 [queue] = self.get_queues()
             except ValueError as e:
                 raise click.ClickException("Queue ID must be specified.") from e
+            except KeyError as e:
+                raise click.ClickException("Could not get queue ID.") from e
```
Comment (KPilnacek) [inline elisctl/lib/api_client.py]: I don't think that `KeyError` can be raised in here. I guess, that [line 41 of `connector`](https://github.com/rossumai/elisctl/pull/75/files#diff-77d6c4254a50ad180ba59e1072305bddR41) module should have been wrapped in this `try ... except`.

If I'm mistaken, please provide a test to cover this part of code.

File: tests/test_connector.py [hunk 7]
```diff
@@ -73,7 +73,10 @@ def test_missing_queue_id(self, mock_token, requests_mock, cli_runner):
...
         requests_mock.get(
             QUEUES_URL,
-            json={"pagination": {"total": 1, "next": None}, "results": [{"id": QUEUE_ID}]},
+            json={
+                "pagination": {"total": 1, "next": None},
+                "results": [{"id": QUEUE_ID, "url": DEFAULT_QUEUE_URL}],
+            },
         )
 
         requests_mock.get(
```
Comment (KPilnacek) [inline tests/test_connector.py]: this mock is not necessary anymore

**Comment by KPilnacek** [comment]
LGTM 🙂 .

**Review by KPilnacek — APPROVED** [review]
LGTM, now, there are some merge conflicts. Please rebase and squash the branch before merging.

**Pull request merged**